server/channels/app/user.go

```diff
@@ -1220,6 +1220,37 @@ func (a *App) UpdateUser(c request.CTX, user *model.User, sendNotifications bool
 		}
 	}
 
+	updateAt := model.GetMillis()
+	if user.AuthService != "" && user.AuthData != nil {
+		switch user.AuthService {
+		case model.UserAuthServiceSaml:
+			user.AuthService = model.UserAuthServiceSaml
```

Comment:
This assignment doesn't make sense as it will go inside the case if the values match.